webpack/assets/javascripts/react_app/routes/index.js

```diff
@@ -1,11 +1,26 @@
-import React from 'react';
-import { Switch, Route } from 'react-router-dom';
+import React, { useEffect } from 'react';
+import { Switch, Route, useLocation } from 'react-router-dom';
 import URI from 'urijs';
 import { routes } from './routes';
 
 let currentPath = null;
 
 const AppSwitcher = () => {
+  const location = useLocation();
```

Comment:
AFAIK with the connected router #7186 , we already gain location as one source of truth via redux

Reply:
I agree, lets use the same source for that